boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -96,6 +96,12 @@ NM = $(CROSSDEV)nm
 OBJCOPY = $(CROSSDEV)objcopy
 OBJDUMP = $(CROSSDEV)objdump
 
+# clang based native compiler doesn't need opjcopy to build simulation.
+# Note: In fact objcopy is not part of MacOS native compiler suite.
+ifneq ("$(shell $(CC) --version | grep clang)","")
+  OBJCOPY = :
```

Comment:
how about we guard the follow code directly:
https://github.com/apache/incubator-nuttx/blob/master/arch/sim/src/Makefile#L288-L292

Reply:
That would work, too. If you prefer that solution, I'll update PR. Pro would be, that it's directly at the location, which is not needed. Con, that OBJCOPY is defined although it's not available. I don't mind, both will working have their Charme.

Reply:
whether OBJCOPY exist depends on user setup, but your finding verify that the renaming process is unnecessary for clang. This is important to explicitly state your finding in Makefile since many people(even me and @yamt) don't know that the clang/gcc work very differently in this case.

Reply:
Got your point ... makes sense.